### ISSUE TITLE
Not able to install from binary

### ISSUE THREAD

**Issue by andrybicio** [hadolint/hadolint#899]
- [x] This is a bug report

My OS:
- Ubuntu 20.04LTS in WSL2 (Windows 10)

```
$  stack ghc -- --version
The Glorious Glasgow Haskell Compilation System, version 9.0.2
```

I wanted to compile and install hadolint on my machine, via the instructions provided in the README. 
```
git clone https://github.com/hadolint/hadolint \
  && cd hadolint \
  && stack install
```
However, the last command (i.e., `stack install`) returns:
```
   Error parsing targets: The specified targets matched no packages.
   Perhaps you need to run 'stack init'?
```

Doing `stack init` as suggested, leads to:
```
Looking for .cabal or package.yaml files to use to init the project.
Using cabal packages:
- ./

(OMITTED)

* Partially matches https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/lts/3/22.yaml
    HsYAML not found
        - hadolint requires >=0
    ShellCheck not found
        - hadolint requires >=0.8.0
    colourista not found
        - hadolint requires >=0
    deepseq version 1.4.1.1 found
        - hadolint requires >=1.4.4 && <1.5
    directory version 1.2.2.0 found
        - hadolint requires >=1.3.0
    gitrev version 1.1.0 found
        - hadolint requires >=1.3.1
    hspec version 2.1.10 found
        - hadolint requires >=2.8.3
    ilist not found
        - hadolint requires >=0
    language-docker not found
        - hadolint requires >=11.0.0 && <12
    megaparsec not found
        - hadolint requires >=9.0.0
    optparse-applicative version 0.11.0.2 found
        - hadolint requires >=0.14.0
    parsec version 3.1.9 found
        - hadolint requires >=3.1.14
    prettyprinter not found
        - hadolint requires >=1.7.0
    Using package flags:
        - hadolint: FlagName "static" = True

Selected resolver: https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/nightly/2022/11/8.yaml
Resolver 'https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/nightly/2022/11/8.yaml' does not have all the packages to match your requirements.
    colourista not found
        - hadolint requires >=0
    Using package flags:
        - hadolint: FlagName "static" = True

This may be resolved by:
    - Using '--omit-packages' to exclude mismatching package(s).
    - Using '--resolver' to specify a matching snapshot/resolver
```

Where the OMITTED part, to my opinion, refers to looking at several `*.yaml* file in different locations and checking that none of them contains all the packages to match my requirements,

**Comment by m-ildefons** [comment]
Stack has been silently removed by @lorenzo in [this](https://github.com/hadolint/hadolint/commit/12473f0317f35fb685c19caaac8a253d187a99c9) commit. The instructions still need to be updated, but a `cabal configure` and `cabal build` should work, provided you have GHC >= 9.2.x. I haven't tested it though but that is what the CI pipeline does.
My development environment sadly is now broken too.

**Comment by andrybicio** [comment]
Thank you for your quick response.
Indeed, running in order 
`cabal configure`
`cabal build`
`cabal install`
 Installed successfully hadolint.

**Comment by lorenzo** [comment]
@m-ildefons do you need help with getting your environment to work again?

**Comment by m-ildefons** [comment]
Thanks for the offer, @lorenzo I got it working and updated the readme. Also thanks for providing the stack config, @mjsteinbaugh .
One problem with stack is that the resolvers can make it quite hard to get a working environment. For example, I tried and failed a couple of months ago to get Hadolint to compile under GHC 9.x, because at the time the resolved didn't have all the requirements I needed. The stable resolvers also take a long time to pick up on changes in `language-docker`, which may slow down development here when fixes in the parser are needed, so I'm reluctant to add stack support back into the project.
